File: src/main/java/org/jhapy/i18n/repository/ActionLookupRepository.java
package org.jhapy.i18n.repository;

import org.jhapy.i18n.domain.ActionLookup;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface ActionLookupRepository extends JpaRepository<ActionLookup, UUID> {
  Optional<ActionLookup> findByActionIdOrName(UUID actionId, String name);

  Optional<ActionLookup> findByName(String name);
}

File: src/main/java/org/jhapy/i18n/repository/ActionLookupRepository.kt
package org.jhapy.i18n.repository

import org.jhapy.i18n.domain.ActionLookup
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface ActionLookupRepository : JpaRepository<ActionLookup, UUID> {
    fun findByActionIdOrName(actionId: UUID, name: String): Optional<ActionLookup>
    fun findByName(name: String): Optional<ActionLookup>
}